torch/csrc/jit/passes/prepack_folding.cpp

```diff
@@ -0,0 +1,67 @@
+#include <stack>
+
+#include <torch/csrc/jit/api/module.h>
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/prepack_folding.h>
+
+namespace torch {
+namespace jit {
+
+namespace {
+bool nodeMatchesPackingOps(Node* n,
+    const std::unordered_set<std::string>& op_white_list) {
+  return (op_white_list.count(n->kind().toQualString()) != 0);
+}
+} // namespace
+
+// Must run this pass after constant folding.
+void FoldPrePackingOps(script::Module& m,
+    const std::unordered_set<std::string>& foldable_prepacking_ops) {
```

Comment:
can this take a function argument of Node* -> bool which can check if a given node is a packing op?

Reply:
Sure, I can do that. I assume you want to pack not only by name, but maybe if certain conditions satisfy.

Reply:
we might just be looking for name as well, it's just taking functions is more general